docs/reference/search/search-your-data/learning-to-rank-model-training.asciidoc

```diff
@@ -0,0 +1,166 @@
+[[learning-to-rank-model-training]]
+=== Deploy and manage Learning To Rank models
+++++
+<titleabbrev>Deploy and manage LTR models</titleabbrev>
+++++
+
+preview::["The Learning To Rank feature is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but this feature is not subject to the support SLA of official GA features."]
+
+NOTE: This feature was introduced in version 8.12.0 and is only available to certain subscription levels.
+For more information, see {subscriptions}.
+
+[discrete]
+[[learning-to-rank-model-training-workflow]]
+==== Train and deploy a model using Eland
+
+Typically, the https://xgboost.readthedocs.io/en/stable/[XGBoost^] model training process uses standard Python data science tools like Pandas and scikit-learn.
+
+
+We have developed an
+https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/08-learning-to-rank.ipynb[example
+notebook^] available in the `elasticsearch-labs` repo. This interactive Python notebook
+details an end-to-end model training and deployment workflow.
+
+We highly recommend using https://eland.readthedocs.io/[eland^] in your workflow, because it provides important functionalities for working with LTR in {es}. Use eland to:
+
+* Configure feature extraction
+
+* Extract features for training
+
+* Deploy the model in {es}
+
+[discrete]
+[[learning-to-rank-model-training-feature-definition]]
+===== Configure feature extraction in Eland
+
+Feature extractors are defined using templated queries. https://eland.readthedocs.io/[Eland^] provides the `eland.ml.ltr.QueryFeatureExtractor` to define these feature extractors directly in Python:
+
+[source,python]
+----
+from eland.ml.ltr import QueryFeatureExtractor
+
+feature_extractors=[
+    # We want to use the score of the match query for the title field as a features:
+    QueryFeatureExtractor(
+        feature_name="title_bm25",
+        query={"match": {"title": "{{query}}"}}
+    ),
+    # We can use a script_score query to get the value of the field rating directly as a feature:
```

Comment:
Consider shortening this comment or breaking into 2 lines so the code snippet window doesn't scroll.